add megalinter and update dependencies

grmbyrn opened the pull request:
* Add [MegaLinter](https://megalinter.io/latest/config-variables-security/)
* Update:
     * TypeScript
     * Prettier
     * Sass
     * Shiki

grmbyrn committed:
add megalinter and update dependencies

grmbyrn committed:
Edit title

grmbyrn committed:
fix errors in megalinter

josecelano commented on .github/workflows/deploy.yml:
```diff
@@ -33,10 +33,6 @@ jobs:
-      name: github-pages
-      url: ${{ steps.deployment.outputs.page_url }}
-
     steps:
```
Hi @grmbyrn why did you remove the environment?

grmbyrn replied:
Hi @josecelano I was getting an error from github-pages.
```
Value 'github-pages' is not valid
The name of the environment used by the job.

Available expression contexts: github, inputs, vars, needs, strategy, matrix
```
I took it out to see how it was and forgot to put it back.

grmbyrn replied:
Hi @josecelano `github-pages` was giving me this error:

```
Value 'github-pages' is not valid
The name of the environment used by the job.

Available expression contexts: github, inputs, vars, needs, strategy, matrix
```

I asked ChatGPT for advice and it recommended removing the environment as it's not necessary for deploying to GitHub Pages. It said that if you want to specify an environment, you can define it separately and reference it within your workflow.

josecelano replied:
The environment exists:

![image](https://github.com/torrust/torrust-website/assets/58816/bdfe10b3-cbfd-4bcc-9bdb-39f4f8e5f73d)

Sometimes I also get those warnings, but the env name could be whatever you want.

grmbyrn replied:
Okay, so do you recommend putting the environment back and ignoring the error?

grmbyrn replied:
Hi @josecelano I've been looking for a solution to this and found an [issue](https://github.com/actions/deploy-pages/issues/344) on [GitHub Actions](https://github.com/actions) repo from May about including `github-pages` as a valid option. It's currently been placed in the backlog of projects. Do you think it's okay to put it back in and ignore the error?

josecelano replied:
> Hi @josecelano I've been looking for a solution to this and found an [issue](https://github.com/actions/deploy-pages/issues/344) on [GitHub Actions](https://github.com/actions) repo from May about including `github-pages` as a valid option. It's currently been placed in the backlog of projects. Do you think it's okay to put it back in and ignore the error?

Yes, @graeme, I would put it back, and I would open a new issue to inform other contributors they could have that error until that schema is updated. We can close it when it is updated, since I think we can't do anything else.

josecelano reviewed (changes requested):
Hi @grmbyrn the PR looks good but there are a couple of suggestions:

### Update dependencies

This could be done in an independent PR unless you need to update them for a feature included in the PR. It's OK to do it in the same PR if you are sure the PR will be merged soon and it's not complex.

### Tagline

I think now it doesn't make sense to use 2 lines:

![image](https://github.com/torrust/torrust-website/assets/58816/f06ba545-8044-4943-821f-2e5a4161edcd)

Maybe we can make the min width longer:

![image](https://github.com/torrust/torrust-website/assets/58816/343977d4-d89a-4641-a0be-d9f71de04bf0)

What do you think?

### Megalinter

I think we should enable Megalinter and fix megalinter errors in the same PR otherwise we are going to have a failing workflow until we fix all errors.

You can run megalinter locally with:

```
npx mega-linter-runner
```

![image](https://github.com/torrust/torrust-website/assets/58816/fc4e4f82-5853-4a7f-89c8-f8e7bab133a3)

MegaLinter generates a local text report so you can fix all of them.

You should use the latest MegaLinter: https://github.com/oxsecurity/megalinter/releases/tag/v7.12.0

grmbyrn commented:
> Hi @grmbyrn the PR looks good but there are a couple of suggestions:
> 
> ### Update dependencies
> This could be done in an independent PR unless you need to update them for a feature included in the PR. It's OK to do it in the same PR if you are sure the PR will be merged soon and it's not complex.

Hi @josecelano okay, I'll remember to dependencies separately unless necessary.

> 
> ### Tagline
> I think now it doesn't make sense to use 2 lines:
> 
> ![image](https://private-user-images.githubusercontent.com/58816/336774921-f06ba545-8044-4943-821f-2e5a4161edcd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc1ODIwOTYsIm5iZiI6MTcxNzU4MTc5NiwicGF0aCI6Ii81ODgxNi8zMzY3NzQ5MjEtZjA2YmE1NDUtODA0NC00OTQzLTgyMWYtMmU1YTQxNjFlZGNkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA1VDEwMDMxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIyNDZjZDIzMTI3MGJjZWNkNDk1OTU3MzBhMjdhNTY3ZGZmMDFhMjMyYmU1NGI2NTg1OGYyYTFlNjMyN2ExYTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.BV7BMeprWeGkgqmy08-glLFw46ZYNFDBD_AU_ftPPho)
> 
> Maybe we can make the min width longer:
> 
> ![image](https://private-user-images.githubusercontent.com/58816/336774804-343977d4-d89a-4641-a0be-d9f71de04bf0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc1ODIwOTYsIm5iZiI6MTcxNzU4MTc5NiwicGF0aCI6Ii81ODgxNi8zMzY3NzQ4MDQtMzQzOTc3ZDQtZDg5YS00NjQxLWEwYmUtZDlmNzFkZTA0YmYwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA1VDEwMDMxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZlMjE1OTVjZWY4ZjVjOWI0YjA1ZjQ1ZTg1MGQ0OTllZjJjMzU0ZmU2ZDRjYjViYzMwMDkwNGEwYzEzZmFmYjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.7G4dyLQbe1cPIz3OGbqvUIsoqy4LRBP-YRLXYJmbGVs)
> 
> What do you think?

Yes, I agree that this looks better on one line and can fix that.

> 
> ### Megalinter
> I think we should enable Megalinter and fix megalinter errors in the same PR otherwise we are going to have a failing workflow until we fix all errors.
> 
> You can run megalinter locally with:
> 
> ```
> npx mega-linter-runner
> ```
> 
> ![image](https://private-user-images.githubusercontent.com/58816/336778911-fc4e4f82-5853-4a7f-89c8-f8e7bab133a3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc1ODIwOTYsIm5iZiI6MTcxNzU4MTc5NiwicGF0aCI6Ii81ODgxNi8zMzY3Nzg5MTEtZmM0ZTRmODItNTg1My00YTdmLTg5YzgtZjhlN2JhYjEzM2EzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA1VDEwMDMxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA2ZmY2YjlhY2I4MTFmNjVmZDRhOWFlZDNhZmFjMzQwMzZmNjY1MTE5ZGYyMTZmOWViNzcyOWRhODAyMWI1MTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ixnp_q_W_HdLFMrZSGYf_l1ITs6bDGYafENnGBJLbhE)
> 
> MegaLinter generates a local text report so you can fix all of them.
> 
> You should use the latest MegaLinter: https://github.com/oxsecurity/megalinter/releases/tag/v7.12.0

Okay, should I fix megalinter errors in this PR?

josecelano commented:
> > Hi @grmbyrn the PR looks good but there are a couple of suggestions:
> > ### Update dependencies
> > This could be done in an independent PR unless you need to update them for a feature included in the PR. It's OK to do it in the same PR if you are sure the PR will be merged soon and it's not complex.
> 
> Hi @josecelano okay, I'll remember to dependencies separately unless necessary.
> 
> > ### Tagline
> > I think now it doesn't make sense to use 2 lines:
> > ![image](https://private-user-images.githubusercontent.com/58816/336774921-f06ba545-8044-4943-821f-2e5a4161edcd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc1ODIwOTYsIm5iZiI6MTcxNzU4MTc5NiwicGF0aCI6Ii81ODgxNi8zMzY3NzQ5MjEtZjA2YmE1NDUtODA0NC00OTQzLTgyMWYtMmU1YTQxNjFlZGNkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA1VDEwMDMxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIyNDZjZDIzMTI3MGJjZWNkNDk1OTU3MzBhMjdhNTY3ZGZmMDFhMjMyYmU1NGI2NTg1OGYyYTFlNjMyN2ExYTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.BV7BMeprWeGkgqmy08-glLFw46ZYNFDBD_AU_ftPPho)
> > Maybe we can make the min width longer:
> > ![image](https://private-user-images.githubusercontent.com/58816/336774804-343977d4-d89a-4641-a0be-d9f71de04bf0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc1ODIwOTYsIm5iZiI6MTcxNzU4MTc5NiwicGF0aCI6Ii81ODgxNi8zMzY3NzQ4MDQtMzQzOTc3ZDQtZDg5YS00NjQxLWEwYmUtZDlmNzFkZTA0YmYwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA1VDEwMDMxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZlMjE1OTVjZWY4ZjVjOWI0YjA1ZjQ1ZTg1MGQ0OTllZjJjMzU0ZmU2ZDRjYjViYzMwMDkwNGEwYzEzZmFmYjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.7G4dyLQbe1cPIz3OGbqvUIsoqy4LRBP-YRLXYJmbGVs)
> > What do you think?
> 
> Yes, I agree that this looks better on one line and can fix that.
> 
> > ### Megalinter
> > I think we should enable Megalinter and fix megalinter errors in the same PR otherwise we are going to have a failing workflow until we fix all errors.
> > You can run megalinter locally with:
> > ```
> > npx mega-linter-runner
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ![image](https://private-user-images.githubusercontent.com/58816/336778911-fc4e4f82-5853-4a7f-89c8-f8e7bab133a3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc1ODIwOTYsIm5iZiI6MTcxNzU4MTc5NiwicGF0aCI6Ii81ODgxNi8zMzY3Nzg5MTEtZmM0ZTRmODItNTg1My00YTdmLTg5YzgtZjhlN2JhYjEzM2EzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA1VDEwMDMxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA2ZmY2YjlhY2I4MTFmNjVmZDRhOWFlZDNhZmFjMzQwMzZmNjY1MTE5ZGYyMTZmOWViNzcyOWRhODAyMWI1MTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ixnp_q_W_HdLFMrZSGYf_l1ITs6bDGYafENnGBJLbhE)
> > MegaLinter generates a local text report so you can fix all of them.
> > You should use the latest MegaLinter: https://github.com/oxsecurity/megalinter/releases/tag/v7.12.0
> 
> Okay, should I fix megalinter errors in this PR?

Hi @grmbyrn yes, if you like. But I will probably take you long. It's usually hard to fix all the MegaLinter errors if it's not installed from the first commit. I would open a new PR just for that.

grmbyrn committed:
Merge branch 'develop' into dependencies

grmbyrn commented:
> > > Hi @grmbyrn the PR looks good but there are a couple of suggestions:
> > > ### Update dependencies
> > > This could be done in an independent PR unless you need to update them for a feature included in the PR. It's OK to do it in the same PR if you are sure the PR will be merged soon and it's not complex.
> > 
> > 
> > Hi @josecelano okay, I'll remember to dependencies separately unless necessary.
> > > ### Tagline
> > > I think now it doesn't make sense to use 2 lines:
> > > ![image](https://private-user-images.githubusercontent.com/58816/336774921-f06ba545-8044-4943-821f-2e5a4161edcd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc1ODIwOTYsIm5iZiI6MTcxNzU4MTc5NiwicGF0aCI6Ii81ODgxNi8zMzY3NzQ5MjEtZjA2YmE1NDUtODA0NC00OTQzLTgyMWYtMmU1YTQxNjFlZGNkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA1VDEwMDMxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIyNDZjZDIzMTI3MGJjZWNkNDk1OTU3MzBhMjdhNTY3ZGZmMDFhMjMyYmU1NGI2NTg1OGYyYTFlNjMyN2ExYTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.BV7BMeprWeGkgqmy08-glLFw46ZYNFDBD_AU_ftPPho)
> > > Maybe we can make the min width longer:
> > > ![image](https://private-user-images.githubusercontent.com/58816/336774804-343977d4-d89a-4641-a0be-d9f71de04bf0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc1ODIwOTYsIm5iZiI6MTcxNzU4MTc5NiwicGF0aCI6Ii81ODgxNi8zMzY3NzQ4MDQtMzQzOTc3ZDQtZDg5YS00NjQxLWEwYmUtZDlmNzFkZTA0YmYwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA1VDEwMDMxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZlMjE1OTVjZWY4ZjVjOWI0YjA1ZjQ1ZTg1MGQ0OTllZjJjMzU0ZmU2ZDRjYjViYzMwMDkwNGEwYzEzZmFmYjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.7G4dyLQbe1cPIz3OGbqvUIsoqy4LRBP-YRLXYJmbGVs)
> > > What do you think?
> > 
> > 
> > Yes, I agree that this looks better on one line and can fix that.
> > > ### Megalinter
> > > I think we should enable Megalinter and fix megalinter errors in the same PR otherwise we are going to have a failing workflow until we fix all errors.
> > > You can run megalinter locally with:
> > > ```
> > > npx mega-linter-runner
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > ![image](https://private-user-images.githubusercontent.com/58816/336778911-fc4e4f82-5853-4a7f-89c8-f8e7bab133a3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc1ODIwOTYsIm5iZiI6MTcxNzU4MTc5NiwicGF0aCI6Ii81ODgxNi8zMzY3Nzg5MTEtZmM0ZTRmODItNTg1My00YTdmLTg5YzgtZjhlN2JhYjEzM2EzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA1VDEwMDMxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA2ZmY2YjlhY2I4MTFmNjVmZDRhOWFlZDNhZmFjMzQwMzZmNjY1MTE5ZGYyMTZmOWViNzcyOWRhODAyMWI1MTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ixnp_q_W_HdLFMrZSGYf_l1ITs6bDGYafENnGBJLbhE)
> > > MegaLinter generates a local text report so you can fix all of them.
> > > You should use the latest MegaLinter: https://github.com/oxsecurity/megalinter/releases/tag/v7.12.0
> > 
> > 
> > Okay, should I fix megalinter errors in this PR?
> 
> Hi @grmbyrn yes, if you like. But I will probably take you long. It's usually hard to fix all the MegaLinter errors if it's not installed from the first commit. I would open a new PR just for that.

Okay, I'll open a new PR for the MegaLinter errors. Is there anything else to fix before merging this PR? The environment in `deploy.yml` is still giving me the error I mentioned.

grmbyrn committed:
Update deploy.yml

Add back in removed environment. Error being shown raised in GitHub Actions with this issue: https://github.com/actions/deploy-pages/issues/344
